src/common/filter/FunctionManager.cpp

```diff
@@ -553,6 +553,75 @@ FunctionManager::FunctionManager() {
             }
         };
     }
+    {
+        auto &attr = functions_["concat_ws"];
+        attr.minArity_ = 2;
+        attr.maxArity_ = 2;
+        attr.body_ = [] (const auto &args) {
+            auto sep   = Expression::asString(args[0]);
+            auto set   = Expression::asString(args[1]);
+
+            const char *p = set.data();
+            size_t size = set.size();
+            size_t offset = 0;
+            std::string s;
+
+            if (size != 0) {
+                uint8_t type = p[0];
```

Comment:
You need to check the `set` value, it may be an empty string or it not the type of data you expect.